src/DacpacTool/ScriptParser.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Reflection;
+using Microsoft.SqlTools.ServiceLayer.BatchParser;
+using Microsoft.SqlTools.Extensibility;
+
+namespace MSBuild.Sdk.SqlProj.DacpacTool
+{
+    public class ScriptParser : ICommandHandler
+    {
+        private readonly Parser _parser;
+        private bool _parsed;
+        private readonly List<string> _includedFileNames = new List<string>();
+        private StringBuilder _scriptBuilder = new StringBuilder();
+
+        public ScriptParser(string sourceFile, IVariableResolver variableResolver)
+        {
+            _parser = new Parser(this, variableResolver, new StreamReader(sourceFile), sourceFile);
+        }
+
+        private void Parse()
+        {
+            if (! _parsed)
+            {
+                _parser.Parse();
+                _parsed = true;
+            }
+        }
+
+        public List<string> CollectFileNames()
```

Comment:
We might want to change this to `IEnumerable<string>` as well.

Reply:
Done